src/LiveComponent/assets/src/ComponentRegistry.ts

```diff
@@ -0,0 +1,35 @@
+import Component from './Component';
+import { getElementAsTagText } from './dom_utils';
+
+const ComponentRegistry = class {
+    private components = new Map<HTMLElement, Component>();
+
+    public registerComponent(element: HTMLElement, definition: Component) {
+        this.components.set(element, definition);
+    }
+
+    public unregisterComponent(element: HTMLElement) {
+        this.components.delete(element);
+    }
+
+    public getComponent(element: HTMLElement): Promise<Component> {
+        return new Promise((resolve, reject) => {
+            let count = 0;
+            const maxCount = 10;
+            const interval = setInterval(() => {
+                const component = this.components.get(element);
+                if (component) {
+                    resolve(component);
+                }
+                count++;
+
+                if (count > maxCount) {
+                    clearInterval(interval);
+                    reject(new Error(`Component not found for element ${getElementAsTagText(element)}`));
+                }
+            }, 5);
```

Comment:
Yea... this might (?) be a different thing? First, about the Livewire events system, it's not something I've added in any way to LiveComponents yet. However, I assumed it was likely that it would be needed at some point.

> What do you think about creating a message bus and dispatch events on it if something happens, so we can register listeners on a global bus related to live components, instead of waiting for something to happen. Something like this:
>
> ```
> ComponentRegistry.hook('live:connect', (component) => {})
> ```

Two possible problems here. I'm not trying to shoot down your idea at all - just keep the conversation flowing. What I have in this PR might, indeed, not be the best solution. Anyways, the 2 possible problems:

A) How would you use this? For example, I believe your callback would be designed to be called when EVERY live component is connected. How would you figure out which `Component` is the correct one that you want if this is called multiple times? Perhaps this is where you're thinking a component name could come in handy? But if this code live inside of a Stimulus controller, perhaps it still makes more sense to match by element - e.g.

```
ComponentRegistry.hook('live:connect', (component) => {
    if (component.element === this.element) {
        // this is the one I want
    }
});
```

B) I think we still have the ordering problem. If you register this hook too later (after the LiveComponent has been connected), then you still won't be called.

Thanks for the convo!

Reply:
Ohh, yes, you are totally right. Livewire is loaded differently, so something is working for them doesn't mean it will work for us too. This hook idea will not solve the problem at hand, so never mind. Your solution is simple and solves the problem :+1: